### PR TITLE
fix: bestEffortFIFO Dump() should return all workloads

### PR DESCRIPTION
**Pull request by kerthcet**
Signed-off-by: kerthcet <kerthcet@gmail.com>

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
I don't know whether this is on purpose as @alculquicondor has fixed the bug in method `Pending()`
#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Review by alculquicondor — COMMENTED** [review]
@denkensk I think this PR kind of makes sense, but then we need another method for just the active workloads.

I guess we can just close this PR. WDYT?

**Comment by kerthcet** [comment]
> @denkensk I think this PR kind of makes sense, but then we need another method for just the active workloads.
> 
> I guess we can just close this PR. WDYT?

em.... I can't get your point. You mean this pr makes sense but just close it.

File: pkg/queue/cluster_queue_best_effort_fifo_test.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func TestClusterQueueBestEffortFIFO(t *testing.T) {
 		"re-queue inadmissible workload": {
 			workloadsToAdd:                 []*kueue.Workload{workloads[0]},
 			inadmissibleWorkloadsToRequeue: []*workload.Info{workload.NewInfo(workloads[1])},
-			wantActiveWorkloads:            sets.NewString(workloads[0].Name),
+			wantActiveWorkloads:            sets.NewString(workloads[0].Name, workloads[1].Name),
```
[the remark this answers is not in the document]
Comment (denkensk) [reply pkg/queue/cluster_queue_best_effort_fifo_test.go:64]: Also workloads[1] is not the active workload here.

**Comment by kerthcet** [comment]
Semantically, I think `dump` should contain all workloads, I talked with @ahg-g before https://github.com/kubernetes-sigs/kueue/pull/248#discussion_r869770945.

Refer to the `manager.Dump()`: https://github.com/kubernetes-sigs/kueue/blob/68ae767b6da43ff376e4e381739cc8152f0da884/pkg/queue/manager.go#L453-L471
 I haven't figured out the test intension here, we want to get all workloads or only admissibleWorkloads, if we only want to get the whole workloads, then we can't change `StrictFIFO` to `BestEffortFIFO` here https://github.com/kubernetes-sigs/kueue/blob/68ae767b6da43ff376e4e381739cc8152f0da884/pkg/scheduler/scheduler_test.go#L55-L75

If there's no problem here, I'm ok to close this pr.

**Comment by alculquicondor** [comment]
The intention of the tests you are trying to modify is actually to get only the admissibleWorkloads, not all of them.

Again, we can reassess if there is a clear need. #248 can simply check the Pending workloads, which is the total.

**Comment by ahg-g** [comment]
Not that this is critical to address, but `Dump` doesn't really mean I want "only active"; the name seems generic enough to indicate that it is a dump of the queue state and so I think it is misleading that it ignores the inadmissible workloads. But sure the signature will need to change to distinguish between the two.

**Pull request closed without merging**